dotcom-rendering/src/web/components/KeyEventsCarousel.importable.tsx

```diff
@@ -0,0 +1,204 @@
+
+	&:hover,
+	&:focus {
+		background-color: ${neutral[60]};
+
+		svg {
+			fill: ${neutral[100]};
+		}
+	}
+`;
+
+export const KeyEventsCarousel = ({
+	keyEvents,
+	filterKeyEvents,
+	format,
+}: Props) => {
+	const [activeIndex, setActiveIndex] = useState(0);
+	const [carousel, setCarousel] = useState<HTMLElement>();
+	const [cards, setCards] = useState<Element[]>();
+	const cardWidth = 200;
+
+	// we only want to count cards fully in view so we round down.
+	const cardsOnScreen = carousel
+		? Math.floor(carousel.clientWidth / cardWidth)
+		: 0;
+
+	const isFirstCard = activeIndex === cardsOnScreen;
+	const isLastCard = activeIndex === keyEvents.length - 1;
+
+	useEffect(() => {
+		const carouselElement = document.getElementById('key-event-carousel');
+		const cardsArray = Array.from(
+			document.querySelectorAll('#key-event-card'),
+		);
+		if (carouselElement) setCarousel(carouselElement);
+		if (cardsArray) setCards(cardsArray);
+	}, []);
+
+	useEffect(() => {
+		const handleIntersect = (entries: IntersectionObserverEntry[]) => {
+			const entry = entries.find((e) => e.isIntersecting);
+			const index = cards?.findIndex((e) => e === entry?.target);
+			if (index && index !== -1) setActiveIndex(index);
+		};
+
+		const observer = new window.IntersectionObserver(handleIntersect, {
+			root: carousel,
+			rootMargin: '0% 0% 0% -100%',
+		});
+
+		cards?.forEach((el) => {
+			observer.observe(el);
+		});
+
+		return () => observer.disconnect();
+	}, [activeIndex, cards, carousel]);
+
+	const goPrevious = () => {
+		if (carousel && !isFirstCard) carousel.scrollLeft -= cardWidth;
+	};
+
+	const goNext = () => {
+		if (carousel && !isLastCard) carousel.scrollLeft += cardWidth;
+	};
+
+	const transformedKeyEvents = keyEvents
+		.filter((keyEvent) => {
+			return keyEvent.title && keyEvent.blockFirstPublished;
+		})
+		.map((keyEvent) => {
+			return {
+				text: keyEvent.title || '', // We fallback to '' here purely to keep ts happy
+				url: `?filterKeyEvents=${filterKeyEvents}&page=with:block-${keyEvent.id}#block-${keyEvent.id}`,
+				date: new Date(keyEvent.blockFirstPublished || ''), // We fallback to '' here purely to keep ts happy
+			};
+		});
+	return (
+		<div id="key-event-carousel" css={carouselStyles}>
+			<span css={containerStyles}>
+				{transformedKeyEvents.map((keyEvent) => {
+					return (
+						<KeyEventCard
+							text={keyEvent.text}
+							url={keyEvent.url}
+							date={keyEvent.date}
+							format={format}
+						/>
+					);
+				})}
+			</span>
+			<Hide until="desktop">
+				{keyEvents.length > 6 && (
+					<span css={buttonContainerStyles}>
+						<button
+							css={[
+								buttonStyles,
+								isFirstCard && disabledButtonStyles,
+							]}
+							type="button"
+							aria-label="Move key events carousel backwards"
+							onClick={goPrevious}
+						>
+							<SvgChevronLeftSingle />
+						</button>
+						<button
+							css={[
+								buttonStyles,
+								isLastCard && disabledButtonStyles,
```

Comment:
There's an argument we shouldn't style controls as disabled. See this note on this [in Source](https://theguardian.design/2a1e5182b/p/435225-button/t/78e328)

>  “It is a bad practice to disable buttons. Disabling the button prevents our chance to tell the user what is wrong. The user keeps clicking the button and is totally in the dark why nothing happens. Keep the button enabled. Let the user click the button. Then show the message why it can’t proceed.” - [A Definitive Guide to Sensible Form Validations ](https://a%20definitive%20guide%20to%20sensible%20form%20validations/)

One could argue this is specific to forms, and this could well be the case here. Has there already been some UX input on this or is this something we can get help on?

Reply:
This is interesting! At the moment, most of the JS comes from needing to keep track of the index to add disabled styling when we reach either end of the carousel. If we were to remove any feedback that we'd reached the beginning and end of the carousel then we could remove most of the JS in this component (as we don't need to track the index) but if we were to show a different type of feedback (such as a message as referenced above) then we would still need some mechanism (and presumably therefore JS) to keep track of the index.

As for UX input, the disabled styling is certainly in the designs which also reflect how the buttons in the existing carousel work, so moving away from this would probably require some input from either @benwuersching or @zeek01

Reply:
If we weren't disabling the buttons what user feedback would we give? Or is it ever acceptable that an active button just does nothing?

Reply:
And if we are to provide some kind of feedback, I think we would still need some mechanism (probably state and useeffects?) to keep track of the index.

Reply:
Following a chat with UX and UI, its been decided we can move forward here without visual feedback or disabling and strip out most of the js here as this sits above the fold. We should keep an eye on this though that a) this doesn't become an assumed design pattern moving forward - UX and UI were ok with this in isolation, but not for the whole site - and b) that we monitor any user feedback that we receive to make sure we're not having a negative impact.

Reply:
Thanks for summarising this so well Anna, this makes sense and sounds like a reasonable approach.